runbotics/runbotics-orchestrator-ui/src/src-app/components/BotProcessRunner.tsx

```diff
@@ -44,55 +65,66 @@ interface BotProcessRunnerProps {
 // eslint-disable-next-line complexity, max-lines-per-function
 const BotProcessRunner: FC<BotProcessRunnerProps> = ({
     className,
+    hasExecutionInfo,
+    processInstance,
     process,
+    isProcessActive = false,
     onRunClick,
     color = 'secondary',
     variant = 'text',
 }) => {
     const dispatch = useDispatch();
     const { enqueueSnackbar, closeSnackbar } = useSnackbar();
-    const [started, setStarted] = useState(false);
+    const [started, setStarted] = useState(isProcessActive);
     const [isSubmitting, setSubmitting] = useState(false);
     const [loading, setLoading] = useState(false);
     const [modalOpen, setModalOpen] = useState(false);
     const { translate } = useTranslations();
     const hasRunProcessAccess = useFeatureKey([FeatureKey.PROCESS_START]);
-    
+
     const processInstances = useSelector(processInstanceSelector);
     const { activeJobs } = useSelector(schedulerSelector);
-    const { orchestratorProcessInstanceId, processInstance } = processInstances.active;
-    const isRunButtonDisabled = started || isSubmitting || !process.system || !process.botCollection;
+    const activeProcessInstance =
+        processInstance ?? processInstances.active.processInstance;
+    const { orchestratorProcessInstanceId } = processInstances.active;
     const isProcessAttended = process?.isAttended && process?.executionInfo;
     const processName = process?.name;
     const processId = process?.id;
-    
+    const isRunButtonDisabled =
+        started || isSubmitting || !process.system || !process.botCollection;
+    const isRerunButtonDisabled = started || isSubmitting || ((processId === activeJobs[0]?.process.id) && !isProcessActive);
+
     useEffect(() => {
         dispatch(schedulerActions.getActiveJobs());
     }, [dispatch, processId]);
-    
+
     useEffect(() => {
+        if (hasExecutionInfo) return;
```

Comment:
I don't like a lot of stuff regarding the existing implementation of the abort button. Let's schedule a pair-programming session to refactor it. It will ease your implementation as well.